core/overdrive.py

```diff
@@ -431,7 +431,9 @@ def get(
         request_headers = dict(Authorization="Bearer %s" % self.token)
         request_headers.update(extra_headers)
 
-        response: Response = self._do_get(url, request_headers)
+        response: Response = self._do_get(
+            url, request_headers, allowed_response_codes=["2xx", "3xx", "401", "404"]
```

Comment:
This is an interesting change, how did this work before this change? Did it always have many failures?

Reply:
What seemed to be happening was that we'd see a 401 and the retry logic added in the OPDS PR (https://github.com/ThePalaceProject/circulation/pull/123) would cause us to avoid going through the "refresh the OAuth token" logic needed to actually properly handle the 401.

Reply:
Ah ok